src/Controllers/AnnouncementController.php

```diff
@@ -113,7 +112,13 @@ public function delete($id)
         try {
             $user = auth()->user();
 
-            $announcement = Announcement::where('id', $id)
+            $announcement = Announcement::join('badaso_courses', 'badaso_courses.id', '=', 'badaso_announcements.course_id')
+                ->where('badaso_announcements.id', $id)
+                ->where(function ($query) {
+                    $query->where('badaso_announcements.created_by', auth()->user()->id)
+                        ->orWhere('badaso_courses.created_by', auth()->user()->id);
```

Comment:
since you have stored the `$user` variable, you can just put `$user->id` then, will be a bit shorter